vue-app/src/components/UI/RecursiveComponent.vue

```diff
@@ -0,0 +1,34 @@
+<template>
+    <div class="recursive-component">
+        <p :style="{ color: getColor(props.depth) }">
+            {{ props.message }} {{ props.depth }}
+        </p>
+        <RecursiveComponent
+        v-if="props.depth < props.maxDepth"
+        :message = "props.message"
+        :depth = "props.depth + 1"
+        :maxDepth = "props.maxDepth"
+        />
+    </div>
+</template>
+
+<script setup lang="ts">
+import type { RecursiveProps } from "../../models/RecursiveProps"
+import RecursiveComponent from "./RecursiveComponent.vue"
+const props = defineProps<RecursiveProps>();
+
+const getColor = (depth : number): string => {
+    const hue = 146 + (depth * 3);
+    const saturation = 172 + (depth * 3);
+    const lightness = 50;
+
+    return `hsl(${hue}, ${saturation}%, ${lightness}%)`;
+  };
+</script>
+
+<style scoped>
+.recursive-component {
+  text-align: center;
+  font-size: 30px;
```

Comment:
why there is 30px, in react 13px?. It should be in ems/rems

Reply:
changed to rems